File: sorts/bubble_sort.py
import numpy as np

def bubble_sort(arr) : 
	lens = len(arr)-1
	while(lens) :
		j = 0
		while(j<=lens):
			if(j+1<= lens and arr[j] > arr[j+1]) : 
				tmp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = tmp	
			j = j+1	
		print(arr, lens)
		lens = lens-1  
	return arr;	 


arr = [11, 3, 0, 19, 40, 66, 100, 3234, 333, 444, 666, 124, 9, 8, 6, 4, 73]
bubble_sort(arr)
	

File: sorts/bubble_sort.c
#include <stdio.h>


int bubble_sort(int *arr, int len)
{
	int *tarr = arr;
	int i=len-1, j=0;
	for(; i>=0; i--)
	{
		for(j=0; j<=i; j++)
		{
			if(j+1 <= i && tarr[j] > tarr[j+1])
			{
				int tmp = tarr[j];
				tarr[j] = tarr[j+1];
				tarr[j+1] = tmp;				
			}			
		}
		for(int i=0; i<len; i++)
			printf("%d ", tarr[i]);
		printf("\n");
	}

	return 0;
}


int main()
{
	int arr[] = {10, -4 ,4, 5, 6, 7, 1, 0 ,9, -3};
	
	int len = sizeof(arr)/sizeof(int);
	bubble_sort((int*)arr, len);

	for(int i=0; i<len; i++)
		printf("%d ", arr[i]);
	printf("\n");

	return 0;
}
